gnocchi/storage/__init__.py

```diff
@@ -254,10 +254,12 @@ def get_cross_metric_measures(metrics, from_timestamp=None,
         for metric in metrics:
             if aggregation not in metric.archive_policy.aggregation_methods:
                 raise AggregationDoesNotExist(metric, aggregation)
-            if (granularity is not None and granularity
-               not in set(d.granularity
-                          for d in metric.archive_policy.definition)):
-                raise GranularityDoesNotExist(metric, granularity)
+            if granularity is not None:
```

Comment:
IIRC it was not because set uses `__hash__` and not `__eq__` for comparison and the `__hash__` function for `SplitKey` was shit. But I fixed it so it might work correctly again?
Anyway my break-based-loop _can_ be faster than using a set that iterate over all archive policy so I ask for a laisser-passer on this one. :-P